File: ruby/problem_solving/codewars/7kyu/string_ends_with.rb
# https://www.codewars.com/kata/51f2d1cafc9c0f745c00037d/train/ruby

def solution(str, ending)
  ending == str.slice(str.size-ending.size..-1)
end


p solution('abcde', 'cde') # true
p solution('abcde', 'abc') # false
p solution('abcde', '') # true

File: go/problem_solving/codewars/7kyu/string_ends_with.go
package main

import "fmt"

func solution(str, ending string) bool {
	if len(ending) > len(str) {
		return false
	}
	return str[len(str)-len(ending):] == ending
}

func main() {
	fmt.Println(solution("abc", "bc") == true)
}
